src/app/icon/almicon.directive.ts

```diff
@@ -8,7 +8,7 @@ import { IconMap } from './iconmap';
 export class AlmIconDirective implements OnInit, OnChanges {
   @Input() 
   iconType: string = 'none';
-
+  iconColor: string = 'none';
```

Comment:
Why do you need a class variable **iconColor** ? You are just using it in **addIcon** function. It can be a local variable, can't it?